node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -295,6 +295,7 @@ impl<C: Context> ActiveValidator<C> {
             panorama = state.panorama().clone();
         }
         let seq_number = panorama.next_seq_num(state, self.vidx);
+        let endorsed = state.endorsements().collect();
```

Comment:
Maybe add a TODO that this is a placeholder: Once the LNC is implemented, only some of our known endorsements will be dependencies of the new vote.

Reply:
I was actually thinking that `state.endorsements` would hold only votes that have to be endorsed according to the `state.panorama` so we would always take all there is and only update the `state.endorsements` when needed.

Reply:
I hadn't thought about it that way… :thinking: 
Not sure that works, though: Endorsement requirements can basically go away again, once we cite a vote that cites the corresponding endorsements… or at least that makes them redundant and a bit wasteful.

Reply:
> once we cite a vote that cites the corresponding endorsements

In this case I would remove those endorsements from `state.endorsements` after adding that vote to this node's protocol state. I will think more about it though.